curriculum/challenges/english/25-front-end-development/workshop-toggle-text-app/67cd61c24bfc0a240132fd2f.md

```diff
@@ -0,0 +1,234 @@
+    data-type="module"
+  >
+    import { ToggleApp } from "./index.jsx";
+    ReactDOM.createRoot(document.getElementById("root")).render(<ToggleApp />);
+  </script>
+</body>
+</html>
+```
+
+```css
+body {
+  font-family: Arial, sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  background-color: #f4f4f4;
+}
+
+#toggle-container {
+  text-align: center;
+  background: white;
+  padding: 20px;
+  border-radius: 10px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+}
+
+#toggle-button {
+  padding: 10px 20px;
+  border: none;
+  background: #007BFF;
+  color: white;
+  border-radius: 5px;
+  cursor: pointer;
+  transition: background 0.3s ease;
+}
+
+#toggle-button:hover {
+  background: #0056b3;
+}
+
+#message {
+  margin-top: 20px;
+  font-size: 1.125rem;
+  color: #333;
+}
+```
+
+```jsx
+import { useState } from "react";
+
+export const ToggleApp = () => {
+  const [isVisible, setIsVisible] = useState(true);
+
+  const handleToggleVisibility = () => {
+    setIsVisible(!isVisible);
+  }
+
+  return (
+    <div id="toggle-container">
+    --fcc-editable-region--
+      <button onClick={handleToggleVisibility} id="toggle-button">Message</button>
+    --fcc-editable-region--
+      {isVisible && <p id="message">I love freeCodeCamp!</p>}
+    </div>
+  );
+};
+```
+
+# --solutions--
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8" />
+  <title>Toggle Visibility</title>
+  <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
+  <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
+  <script 
+    data-plugins="transform-modules-umd"
+    type="text/babel"
+    src="index.jsx"
+  ></script>
+  <link rel="stylesheet" href="styles.css" />
+</head>
+<body>
+  <div id="root"></div>
+  <script
+    data-plugins="transform-modules-umd"
+    type="text/babel"
+    data-presets="react"
+    data-type="module"
+  >
+    import { ToggleApp } from "./index.jsx";
+    ReactDOM.createRoot(document.getElementById("root")).render(<ToggleApp />);
+  </script>
+</body>
+</html>
+```
+
+```css
+body {
+  font-family: Arial, sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  background-color: #f4f4f4;
+}
+
+#toggle-container {
+  text-align: center;
+  background: white;
+  padding: 20px;
+  border-radius: 10px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+}
+
+#toggle-button {
+  padding: 10px 20px;
+  border: none;
+  background: #007BFF;
+  color: white;
+  border-radius: 5px;
+  cursor: pointer;
+  transition: background 0.3s ease;
+}
+
+#toggle-button:hover {
+  background: #0056b3;
+}
+
+#message {
+  margin-top: 20px;
+  font-size: 1.125rem;
+  color: #333;
+}
+```
+
+```jsx
+// Preview doesn't work for this type of import
+// import { useState } from "react"; 
+const { useState } = React;
```

Comment:
Look into getting the other import to work for previews

Reply:
Here are the error messages I get when I try to use `import { useState } from "react"; `

<img width="699" alt="Screenshot 2025-03-12 at 12 46 23 PM" src="https://github.com/user-attachments/assets/5645f8bb-4000-404a-a066-c3efcd0b2936" />

Reply:
For now, we are going to use the `const { useState } = React;` syntax and have that already be imported for campers on the first step